kragent.go

```diff
@@ -93,6 +96,36 @@ func main() {
 				err = errors.New("invalid shutdown mode")
 			}
 			log.Printf("Reboot for mode %q = %v", m.Arguments.Mode, err)
+		case "guest-fsfreeze-freeze":
+			count, err := fsfreezeFreeze()
+			if err != nil {
+				log.Printf("Unable to fsfreeze-freeze the filesystems: %v", err)
+				// On error, all filesystems will be thawed back.
+				// If no filesystems are frozen as a result of this call,
+				// then guest-fsfreeze-status will remain “thawed” and calling guest-fsfreeze-thaw is not necessary.
+				_, err := fsfreezeThaw()
+				if err != nil {
+					log.Printf("Unable to fsfreeze-thaw the previously frozen filesystems after a freeze failure: %v", err)
+				}
+				je.Encode(Return{0})
+				bw.Flush()
```

Comment:
Is the fallthrough intentional here and below? I would have expected that we shouldn’t update the status when the operation fails, no?

Reply:
Oops not intentional, the `{Return}` tricked me good catch! Updated.